### ISSUE TITLE
Possible bug in rgb2ycc.py and/or ycc2rgb.py

### ISSUE THREAD

**Issue by vicente-gonzalez-ruiz** [Sistemas-Multimedia/MCDWT#105]
The process is not reversible and could be. My test has been:

```
python ycc2rgb.py -i ../images/000.png -o /tmp/000.png
python rgb2ycc.py -i /tmp/000.png -o /tmp/000b.png
./show_differences.sh -1 /tmp/000b.png -2 ../images/000.png -o /tmp/diff.png
```

And `/tmp/diff.png` is not zero.

Be careful. It is possible that OpenCV does not provide a lossless process, but please, check it.

**Comment by tukaramhl** [comment]
That's right, the operation is mathematically invertible. It could be a rounding problem with the integers. I'll check this out as soon as possible.

**Comment by tukaramhl** [comment]
I think this is relevant. From the opencv [docs](http://docs.opencv.org/3.1.0/d7/d1b/group__imgproc__misc.html#ga397ae87e1288a81d2363b61574eb8cab):
> If you use cvtColor with 8-bit images, the conversion will have some information lost. For many applications, this will not be noticeable but it is recommended to use 32-bit images in applications that need the full range of colors or that convert an image before an operation and then convert back.

Also, my tests are showing inconsistency when converting a 24-bit image back and forth. I think that the best course of action is to implement our own library, just to be on the safe side.

**Comment by vicente-gonzalez-ruiz** [comment]
Hi @tukaramhl . Thanks for your excellent reply. Yes, it seems to be that OpenCV is *guilty*. If you are interested in fully solved this minimal problem, please, go ahead. However, this is not essential for our project. In fact, in a typical process (MCDWT + entropy coding), we will never have the necessity of traveling between color domains (we will perform all the process in the YCbCr domain). This tool is mainly for visual purposes and in the current state, is fully functional.

So, it is up to you :-)

**Comment by tukaramhl** [comment]
I don't like the idea of leaving the scripts in this state, so I'll look into a new implementation as soon as I can.

**Comment by tukaramhl** [comment]
Just to update this issue. I have tried to load the image, convert it into an array (uint 8) and apply the transformation, but even this produces some data corruption in the image. I will have to look into the numpy arrays for a possible overflow/underflow.

I'll leave a couple of links to add more documentation. It's funny the level of confusion between YUV, YCbCr and YPbPr that exist in many solutions arround the net :).

[Color Conversion](http://www.equasys.de/colorconversion.html)
[SO: Color space mapping](http://stackoverflow.com/questions/34913005/color-space-mapping-ycbcr-to-rgb)

**Comment by vicente-gonzalez-ruiz** [comment]
Right.
